source/common/config/grpc_mux_impl.h

```diff
@@ -32,6 +33,12 @@ class GrpcMuxImpl : public GrpcMux,
                     public GrpcStreamCallbacks<envoy::service::discovery::v3::DiscoveryResponse>,
                     public Logger::Loggable<Logger::Id::config> {
 public:
+  GrpcMuxImpl(const LocalInfo::LocalInfo& local_info, Grpc::RawAsyncClientPtr async_client,
+              Event::Dispatcher& dispatcher, const Protobuf::MethodDescriptor& service_method,
+              envoy::config::core::v3::ApiVersion transport_api_version,
+              Random::RandomGenerator& random, Stats::Scope& scope,
+              const RateLimitSettings& rate_limit_settings, bool skip_subsequent_node,
+              bool enable_type_url_downgrade_and_upgrade);
```

Comment:
I'm not sure if `enable_type_url_downgrade_and_upgrade` is required. Is it not good just to use runtime flag? I think that we should allow fallback behavior whenever the runtime feature is enabled.

Reply:
This constructor is for testing purpose, in unit test we don't have runtime flag.

Reply:
I have removed the constructor with bool set explicitly.